File: src/shared/types/knowledge.ts
export type Knowledge = {
	id: string;
	type_id: string;
	title: string;
	body: string;
	plain_text: string;
	sort_order: number;
	created_at: string;
	updated_at: string;
};

export type KnowledgeOutline = {
	id: string;
	type_id: string;
	title: string;
	sort_order: number;
};

export type KnowledgeSearchResult = {
	id: string;
	type_id: string;
	title: string;
	matched_text: string;
};

export type KnowledgeType = {
	id: string;
	work_id: string;
	name: string;
	color: string | null;
	icon: string | null;
	sort_order: number;
	created_at: string;
	updated_at: string;
};

export type KnowledgeTypeOutline = {
	id: string;
	name: string;
	color: string | null;
	icon: string | null;
	sort_order: number;
	count: number;
};

File: src-tauri/src/models/knowledge.rs
use chrono::NaiveDateTime;
use serde::Serialize;
use sqlx::FromRow;

#[derive(Debug, Serialize, FromRow)]
pub struct Knowledge {
    pub id: String,
    pub type_id: String,
    pub title: String,
    pub body: String,
    pub plain_text: String,
    pub sort_order: i64,
    pub created_at: NaiveDateTime,
    pub updated_at: NaiveDateTime,
}

#[derive(Debug, Serialize, FromRow)]
pub struct KnowledgeOutline {
    pub id: String,
    pub type_id: String,
    pub title: String,
    pub sort_order: i64,
}

#[derive(Debug, Serialize, FromRow)]
pub struct KnowledgeSearchResult {
    pub id: String,
    pub type_id: String,
    pub title: String,
    pub matched_text: String,
}
